src/sonic-frr/patch/0006-BFD-Enhancement.patch

```diff
@@ -0,0 +1,1562 @@
+diff --git a/bfdd/bfd.c b/bfdd/bfd.c
+index fc2d0bea9..abbd0e6a7 100644
+--- a/bfdd/bfd.c
++++ b/bfdd/bfd.c
```

Comment:
Yes. I'd prefer to have them as individual patches.
In this case, though, I'd try to push the changes into FRR itself.

Reply:
This is not a patch anymore. Non bug fixes, must be upstreamed to frr first.

Reply:
All - I think our position here is that if we are required to upstream to FRR first, then this contribution will miss the current cycle (201908). This is because: -
1) We likely won't have time to get it through the FRR community in time
2) Doing so may require us to advance the FRR in SONiC

To avoid this we should proceed with the patch  as presented, and upstream to FRR in parallel. We will then converge in a future FRR release.

Reply:
We should avoid 1500+ lines of functional code changes being treated as a patch. Any change by frr or sonic that interacts with that patch/file, or even changes bfd functionality, will create issues since people (in both frr and sonic) will be unaware of these changes.

Have you at least opened up a PR in frr repo for these changes to get them upstreamed? What's the PR for it?

Reply:
the frr patch lack of description, need to add description, what are the enhancements. Without description, it is hard to know what enhancements are. Then, we should consider add to sonic-frr 7.0.1 branch as a commits. @pavel-shirshov . Meanwhile, I agree with @nikos-github , we need at least open PR for frr and check the response from them.

Reply:
@lguohan as soon as we integrate such patches as part of our frr we're going to stuck with one version of frr. Do we want that? Also as soon as we push the patch to our branch it's going to be our responsibility of fixing issues.

@ben-gale why your patch is critical for Aug branch of sonic? Will aug branch work without it?

Reply:
@pavel-shirshov The patch is needed for us to contribute the BFD feature to the SONiC community. Without it, no feature .....

Reply:
@pavel-shirshov , i think broadcom will maintain the patch. I do not think it will us to fix those issues w.r.t to BFD. @ben-gale, correct?

Reply:
Correct